doc/src/documentation_style_guide.rst

```diff
@@ -0,0 +1,1019 @@
+a math context because it does not render as the Greek letter (α).
+
+Style Preferences
+-----------------
+
+**Spelling and Punctuation**
+
+All narrative writing in SymPy follows American spelling and punctuation
+standards. For example, “color” is preferred over “colour” and commas should be
+placed inside of quotation marks.
+
+**Examples**
+
+::
+
+  If the ``line_color`` aesthetic is a function of arity 1, then the coloring is
+  a function of the x value of a point.
+
+  The term "unrestricted necklace," or "bracelet," is used to imply an object
+  that can be turned over or a sequence that can be reversed.
+
+If there is any ambiguity about the spelling of a word, such as in the case of
+a function named after a person, refer to the spelling of the actual SymPy
+function.
+
+For example, Chebyshev polynomials are named after Pafnuty Lvovich Tchebychev,
+whose name is sometimes transliterated from Russian to be spelled with a “T,”
+but in SymPy it should always be spelled “Chebyshev” to refer to the SymPy
+function.
+
+**Example**
+
+::
+
+  class chebyshevt(OrthogonalPolynomial):
+      r"""
+      Chebyshev polynomial of the first kind, $T_n(x)$
+      ...
+
+      """
+
+**Capitalization**
+
+Title case capitalization is preferred in all SymPy headings.
+
+**Example**
+
+::
+
+  What is Symbolic Computation?
+  -----------------------------
+
+**Grammar**
+
+WORK IN PROGRESS: Google Season of Docs technical writer will track common
+grammar mistakes (e.g., using certain articles correctly) while editing and add
+to this guide.
+
+Tone Preferences
+----------------
+
+Across SymPy documentation, please write in:
+
+* The present tense (e.g., In the following section, we are going to learn...)
+* The first-person inclusive plural (e.g., We did this the long way, but now we
+  can try it the short way...)
+* Use the generic pronoun “you” instead of “one.” Or use “the reader” or “the
+  user.” (e.g., You can access this function by... The user can then access
+  this function by...)
+* Use the gender-neutral pronoun “they” instead of “he” or “she.” (e.g., A good
+  docstring tells the user exactly what they need to know.)
+
+Avoid extraneous or belittling words such as “obviously,” “easily,” “simply,”
+“just,” or “straightforward.”
+
+Avoid unwelcoming or judgement-based phrases like “That is wrong.” Instead use
+friendly and inclusive language like “A common mistake is...”
+
+Avoid extraneous phrases like, “we just have to do one more thing.”
+
+Docstring Guidelines
+====================
+
+To contribute to SymPy’s docstrings, please read these guidelines in full.
+
+A documentation string (docstring) is a string literal that occurs as the first
+statement in a module, function, class, or method definition. Such a docstring
+becomes the ``__doc__`` special attribute of that object.
+
+**Example**
+
+Here is a basic docstring::
+
+  def fdiff(self, argindex=1):
+      """
```

Comment:
If the NumPy docstring standard (https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard) is supposed to be used, there should not be a line break between `"""` and the title line.

Reply:
We aren't following the NumPy standard exactly. IMO the 'no newline after """' rule looks really bad and asymmetrical and makes the docstrings harder to read in the source code.

Reply:
I think we should follow the numpydoc style unless we plan to fork the numpydoc sphinx extension. You may think it looks bad, but others (i.e. numpydoc authors) may not. I think it is generally better to follow a style and deal with minor things that we don't like about it. numpydoc is practically a standard, not just a style. That's what's nice about the philosophy of tools like black, which I think is generally good.

Reply:
Putting a newline after the quotes is also a standard. I've seen it done more often than the other way. 

And I think at the end of the day, we should do what will be the best for the end users consuming the documentation first. If that means breaking with a preexisting standard because the standard says to do something in a poor way, then we should absolutely break the standard. numpydoc is good, but it isn't the epitome of documentation standards.

Also this particular issue is completely irrelevant to the Sphinx extension.

Reply:
(and I also think black is bad so you aren't really selling me there)

Reply:
> Putting a newline after the quotes is also a standard.

What pep or style tool is it encoded in?

> And I think at the end of the day, we should do what will be the best for the end users consuming the documentation first. If that means breaking with a preexisting standard because the standard says to do something in a poor way, then we should absolutely break the standard. numpydoc is good, but it isn't the epitome of documentation standards.

I disagree with this. The issue is that "a poor way" is subjective. There is no objective way to determine what is poor except if you adopt a standard, e.g. numpydoc or pydocstyle or sphinx raises and error, which is objective.

> (and I also think black is bad so you aren't really selling me there)

After watching a decade of time wasted in PRs to sympy dealing with code style comments, I think I'm sold on the philosophy of "just pick and use a standard regardless if you don't like some parts of it". black is just a tool that does that for you. There are close to 800 contributors to this project and they all have different opinions on details of code style. It will always be impossible to have a style that everyone likes. I don't think we need to roll our own specs for the style for code or docs, we should just pick something that exists and follow it. Following numpydoc seems like a no-brainer to me since almost all other packages in the scipy set of packages follow it and we already parse our docs with it. I think that all we need is some elaboration on the recommended class docstring guides because we have many classes that describe objects that represent and act like mathematical things, which are a bit different than most other scipy ecosystem packages' objects.

Reply:
I'm a little confused about the numpydoc because under the documenting classes section I'm seeing an example (class photo) where there is a new line for the text after the triple quotes: https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes

Also, is there a place where they specifically say that there shouldn't be a line break in between the triple quotes and first line of text? I'm not seeing one. Just that a docstring "for consistency, is surrounded by triple double quotes."

Personally, I think the line break between the triple quotes and first line of text makes the docstring easier to read and makes it looked "bookended" by the triple quotes and so tidier. A general rule of thumb in copy editing/writing is that more whitespace increases readability. But I also understand the point of sticking with a current standard that more people are accustomed to doing so that you are not constantly fighting a copy editing battle to keep your docstrings consistent.

Reply:
The `Photo` class is interesting, I think this is an anomaly in the NumPy guide and should be fixed. You should open an issue for that!

> Also, is there a place where they specifically say that there shouldn't be a line break in between the triple quotes and first line of text? I'm not seeing one.

Me neither.

But it's obvious that it has to be one of the two options, it cannot be both.
And since all examples (except the `Photo` class) don't have a newline before the title line, it's quite obvious which one should be used.

But mentioning this explicitly would of course be better. Another good issue for the NumPy project!

Reply:
It's worth noting that this only affects the way the documentation looks in the source code. It doesn't affect the resulting rendering in Sphinx and it doesn't affect the `help()` display either. So in a technical sense, this is a source code formatting issue, not a documentation formatting issue. We don't really have many source formatting rules in SymPy, so I personally don't have any issues with allowing both ways (although I personally prefer the newline after the triple quotes, and my emacs is actually configured to do this automatically).

Reply:
PEP 257 also doesn't say that the newline is not allowed, it says of multi-line docstrings "The summary line may be on the same line as the opening quotes or on the next line." https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

From a copy editing/writing viewpoint of the RST files, the newline makes for better readability, so I agree we shouldn't make a guideline against it here in this guide.